recipes/googleapis/all/conanfile.py

```diff
@@ -126,7 +126,7 @@ def deactivate_library(key):
         #  - Inconvenient macro names from usr/include/sys/syslimits.h in some macOS SDKs: GID_MAX
         #    Patched here: https://github.com/protocolbuffers/protobuf/commit/f138d5de2535eb7dd7c8d0ad5eb16d128ab221fd
         #    as of 3.21.4 issue still exist
-        if Version(self.dependencies["protobuf"].ref.version) <= "3.21.5" and self.settings.os == "Macos" or \
+        if Version(self.dependencies["protobuf"].ref.version) <= "3.21.9" and self.settings.os == "Macos" or \
```

Comment:
Why did this need to change?

I am confused by there's a version check if this was not fixed but the commit is huge so I dont know what this means 😖 

Hopefully you are familiar with this project?

Reply:
sorry, I am not familiar with it😅, from the link in the comment I know there are some bugs on older versions to compile on macOS including 3.21.9, so the original author added the logic.

Reply:
I need to use protobuf/3.21.9 in my project, but it conflicts with googleapi and grpc-proto which depends on protobuf/3.21.4, I am trying to upgrade all of them to protobuf/3.21.9

Reply:
Hi @coryan - thank you for the analysis.

Indeed, I suspect at the time this check was introduced it was not known which (if any) version fixed this fully - hence the need to keep bumping it until the actual minimal version.
The Conan Center protobuf package still does not follow the new versioning, which is indeed confusing. @coryan, the link you have posted would indicate that it is _python_ that changes the version scheme to 4.x, and C++ stays in 3.x? So perhaps the upper bound here should be `3.22.0` (at least in line with Conan Center's protobuf versioning scheme).

On the other hand, we _have_ backported the fix (related to `GID_MAX`) to all current versions of protobuf we support:
https://github.com/conan-io/conan-center-index/pull/16453

so we could probably do away with this check altogether, and actually build these.
What do you think @prince-chrismc , we have two options:
* change this check to use `3.22.0` as the upper bound (if @coryan confirms this is indeed correct for the 3.x series)
* remove the part of this conditional that checks that `self.settings.os` is macOS, and keep it for android only (this would mean that we assume that users are already using a patched protobuf with the changes from https://github.com/conan-io/conan-center-index/pull/16453/files)

Reply:
> the link you have posted would indicate that it is python that changes the version scheme to 4.x, and C++ stays in 3.x? So perhaps the upper bound here should be 3.22.0 (at least in line with Conan Center's protobuf versioning scheme).

If you mean the news from [2022-05-06](https://protobuf.dev/news/2022-05-06/): yes that bumped Python to 4.x, but I intended that only as a reference for the introduction of the new versioning scheme.  The breaking changes for C++ in v22 (turning it into 4.22.x for C++) are in the news from [2022-08-03](https://protobuf.dev/news/2022-08-03/).

Reply:
we have patched this in CCI (eg 3.21.9 and 3.21.4) for portability and it's officially fixed in the 4.x https://protobuf.dev/support/migration/ (which starts at 22)

I think we can accept this change as is with `3.21.9` since that is whats being tested since there are revision which are not update it can be consufing is someone has an older revision, so yes an upper bound of 3.22 👍 

With 4.x they made build script changes so we can just start a refresh recipe folder and drop the logic all together 🤞 so. I think is easier to just leave the logic along for now.